Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -558,30 +559,69 @@ class IntegrationLogger(object):
 
     def __init__(self, ):
         self.messages = []  # type: list
-
-    def __call__(self, message):
+        self.write_buf = []  # type: list
+        self.replace_strs = []  # type: list
+        if (not os.getenv('COMMON_SERVER_NO_AUTO_REPLACE_STRS') and hasattr(demisto, 'getParam')
+                and isinstance(demisto.getParam('credentials'), dict)
+                and demisto.getParam('credentials').get('password')):
+            pswrd = self.encode(demisto.getParam('credentials').get('password'))
+            to_encode = pswrd
+            if IS_PY3:
+                to_encode = pswrd.encode('utf-8', 'ignore')
+            self.add_repalce_strs(pswrd, base64.b64encode(to_encode))
+
+    def encode(self, message):
         try:
-            self.messages.append(str(message))
-
+            res = str(message)
         except UnicodeEncodeError as ex:
             # could not decode the message
             # if message is an Exception, try encode the exception's message
             if isinstance(message, Exception) and message.args and isinstance(message.args[0], STRING_OBJ_TYPES):
-                self.messages.append(message.args[0].encode('utf-8', 'replace'))
+                res = message.args[0].encode('utf-8', 'replace')  # type: ignore
             elif isinstance(message, STRING_OBJ_TYPES):
                 # try encode the message itself
-                self.messages.append(message.encode('utf-8', 'replace'))
+                res = message.encode('utf-8', 'replace')  # type: ignore
             else:
-                self.messages.append("Failed encoding message with error: {}".format(ex))
+                res = "Failed encoding message with error: {}".format(ex)
+        for s in self.replace_strs:
+            res = res.replace(s, '<XX_REPLACED>')
+        return res
+
+    def __call__(self, message):
+        self.messages.append(self.encode(message))
+
+    def add_repalce_strs(self, *args):
```

Comment:
logger docstring (for the script helper) should recommend user to call `add_repalce_strs` with the credentials.

we also need to add it to code standard (next to `handle_proxy`)

Reply:
added comments